Library/Homebrew/os/mac/cmd/config.rb

```diff
@@ -0,0 +1,196 @@
+#:  * `config`:
+#:    Show Homebrew and system configuration useful for debugging. If you file
+#:    a bug report, you will likely be asked for this information if you do not
+#:    provide it.
```

Comment:
That's not pedantry; it's still a bit messy, I agree. This is why I tried to keep this PR a bit smaller; reviewing individual commits should help but once I've nailed the approach it'll be easier to split future things into individual PRs.